shard/src/ext/latex_cog.py

```diff
@@ -19,20 +23,46 @@ def __init__(self, bot):
             s = f.read()
         self.base_tex = Template(s)
 
+        self.illegal_commands = [
+            "\\write",
+            "\\tempfile",
+            "\\openout",
+            "\\newwrite",
+            "\\write",
+            "\\input",
+            "\\usepackage",
+            "\\include"
+        ]
+
     @commands.command(aliases=['tex'])
-    async def latex(self, ctx, latex):
+    @cooldown(2, 15, BucketType.user)
+    async def latex(self, ctx, *latex):
+        for l in latex:
+            for c in self.illegal_commands:
+                if l.find(c) != -1:
+                    await ctx.send("Illegal latex command")
+                    return
+        latex = " ".join(latex)
+
         with tempfile.TemporaryDirectory() as work_dir:
             out_txt = self.base_tex.substitute(my_text=latex)
 
             with open(os.path.join(work_dir, 'out.tex'), 'w') as f:
                 f.write(out_txt)
 
-            tex = await create_subprocess_exec('latex', '-halt-on-error', 'out.tex',
+
+            tex = await create_subprocess_exec('latex', '-halt-on-error', '-no-shell-escape',
+                                               '-interaction batchmode', 'out.tex',
                                                cwd=work_dir,
                                                stdout=DEVNULL,
                                                stderr=DEVNULL,
                                                close_fds=True)
-            await tex.wait()
+
+            try:
+                await wait_for(tex.wait(), timeout=3)
+            except TimeoutError:
```

Comment:
do you still need to manually kill the process in this case?

Reply:
If I'm interpreting the python documentation correctly, if the wait_for function timesout it will cancel the awaitable which should cancel the command.